app/models/work_activity.rb

```diff
@@ -40,15 +40,78 @@ def created_by_user
   end
 
   def message_html
-    # convert user references to user links
-    text = message.gsub(USER_REFERENCE) do |at_uid|
-      uid = at_uid[1..-1]
-      user = User.where(uid: uid).first
-      user_info = user&.display_name_safe || uid
-      "<a class='comment-user-link' title='#{user_info}' href='{USER-PATH-PLACEHOLDER}/#{uid}'>#{at_uid}</a>"
-    end
-    # allow ``` for code blocks (Kramdown only supports ~~~)
-    text = text.gsub("```", "~~~")
-    Kramdown::Document.new(text).to_html
+    if activity_type == "CHANGES"
+      changes_html
+    else
+      comment_html
+    end
   end
+
+  private
+
+    def comment_html
+      # convert user references to user links
+      text = message.gsub(USER_REFERENCE) do |at_uid|
+        uid = at_uid[1..-1]
+        user = User.where(uid: uid).first
+        user_info = user&.display_name_safe || uid
+        "<a class='comment-user-link' title='#{user_info}' href='{USER-PATH-PLACEHOLDER}/#{uid}'>#{at_uid}</a>"
+      end
+      # allow ``` for code blocks (Kramdown only supports ~~~)
+      text = text.gsub("```", "~~~")
+      Kramdown::Document.new(text).to_html
+    end
+
+    # Returns the message formatted to display changes that were logged as an activity
+    def changes_html
+      text = ""
+      changes = JSON.parse(message)
+      changes.keys.each do |field|
+        if changes[field].is_a?(Array)
+          # Multi-value change, process each individual entry
+          values = changes[field].map { |value| change_value_html(value) }
+          text += "<p><b>#{field}</b>:#{values.join}</p>"
```

Comment:
@hectorcorrea was there intention to make the formatting different for a multi-value?  It might be nice to move the formatting out of the if, but adding a space would match it up.
```suggestion
          text += "<p><b>#{field}</b>: #{values.join}</p>"
```

Reply:
This has now been simplified now that we always return an array. Again, thanks for the hint.